deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/logging/DeploymentScannerLogger.java

```diff
@@ -407,4 +407,16 @@
      */
     @Message(id = 36, value = "Deployment model operation failed. %s")
     RuntimeException deployModelOperationFailed(String msg);
+
+    @Message(id = 37, value = "%s does not exist")
+    String directoryIsNonexistent(String path);
+
+    @Message(id = 38, value = "%s is not a directory")
+    String isNotADirectory(String path);
+
+    @Message(id = 39, value = "%s is not writable")
+    String directoryIsNotWritable(String path);
+
+    @Message(id = 40, value = "Could not find a path called '%s'")
+    String pathEntryNotFound(String pathName);
```

Comment:
Messages like these that are only used to create an exception should all the exception type (i.e. OperationFailedException) as the return type for the method, not String. Then in the places that use the methods, just

throw DeploymentScannerLogger.ROOT_LOGGER.pathEntryNotFound(pathName);